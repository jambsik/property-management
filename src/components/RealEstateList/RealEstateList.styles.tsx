export const realEstateListPaperStyles = {
    width: '1200px',
    height: '900px',
};
