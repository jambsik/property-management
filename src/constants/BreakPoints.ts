export enum BreakPoints {
    Xs = 'xs', // extra-small: 0px
    Sm = 'sm', // 600px
    Md = 'md', // 900px
    Lg = 'lg', // 1200px
    Xl = 'xl', // 1536px
}
