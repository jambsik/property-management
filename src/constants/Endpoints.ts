export enum Endpoints {
    RealEstate = 'api/real-estate.json',
    RealEstateFilters = 'api/real-estate-filters.json',
}
