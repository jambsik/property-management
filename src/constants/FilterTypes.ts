export enum FilterTypes {
    String = 'string',
    Number = 'number',
    Year = 'year',
}
