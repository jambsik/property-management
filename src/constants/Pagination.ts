export enum Pagination {
    DefaultPage = 1,
    DefaultLimit = 4,
    offset = 1,
}
