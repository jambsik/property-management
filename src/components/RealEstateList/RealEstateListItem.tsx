import React from 'react';
import { Grid, Typography } from '@mui/material';
import { Box, styled } from '@mui/system';
import { RealEstate } from '../../models/RealEstate';
import { realEstateListItemImgStyles, realEstateListItemRootStyles, realEstateListItemTextStyles } from './RealEstateList.styles';
import { useTranslation } from 'react-i18next';
interface RealEstateListItemProps {
    item: RealEstate;
}

const RealEstateListItem = ({ item }: RealEstateListItemProps) => {
    const { t } = useTranslation();
    const Image = styled('img')``;
    const valuesToShow = ['street', 'yrsold', 'yearbuilt', 'saleprice', 'garagecars', 'kitchenabvgr', 'bedroomabvgr'];

    return (
        <Box sx={realEstateListItemRootStyles}>
            <Image sx={realEstateListItemImgStyles} alt="home-image" src="images/home.svg" />
            <Grid container rowSpacing={1} columnSpacing={{ xs: 2 }}>
                {valuesToShow.map((key: string) => (
                    <React.Fragment key={key}>
                        <Grid item xs={2}>
                            <Typography color="info" variant="body1">
                                {t(`realEstate.filters.${key}`)}:
                            </Typography>
                        </Grid>
                        <Grid item xs={2}>
                            <Typography sx={realEstateListItemTextStyles} color="secondary" variant="body1">
                                {item[key as keyof RealEstate]}
                            </Typography>
                        </Grid>
                    </React.Fragment>
                ))}
            </Grid>
        </Box>
    );
};

export default RealEstateListItem;

/*

                <Box key={key} sx={realEstateListItemBoxStyles}>
                 
                </Box>
*/
