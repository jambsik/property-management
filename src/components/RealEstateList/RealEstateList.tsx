import { IconButton, Pagination, Paper, Typography } from '@mui/material';
import { Box, styled } from '@mui/system';
import React from 'react';
import { useTranslation } from 'react-i18next';
import SearchIcon from '@mui/icons-material/Search';

import {
    realEstateListDelimiterStyles,
    realEstateListIconStyles,
    realEstateListImageStyles,
    realEstateListItemStyles,
    realEstateListLoaderStyles,
    realEstateListPaperStyles,
} from './RealEstateList.styles';
import { Size } from '../../constants/Size';
import { RealEstate } from '../../models/RealEstate';
import RealEstateListItem from './RealEstateListItem';

interface RealEstateListProps {
    page: number;
    numberOfPages?: number;
    items: Array<RealEstate>;
    onDispatchAction: (page?: number) => void;
}

const RealEstateList = ({ items, page, numberOfPages, onDispatchAction }: RealEstateListProps) => {
    const { t } = useTranslation();
    const hasItems = items.length > 0;
    const Image = styled('img')``;
    const Delimiter = styled('div')``;

    const hasPagination = numberOfPages && numberOfPages > 1;

    const onChangePage = (event: React.ChangeEvent<unknown>, page: number) => {
        onDispatchAction(page);
    };
    const getLoaderBlock = () => (
        <>
            <Typography color="info" variant="h4">
                {t('realEstate.list.title')}
            </Typography>
            <Image alt="build" sx={realEstateListImageStyles} src="/images/build.svg" />
            <Typography color="info" variant="h4">
                {t('realEstate.list.subTitle')}
            </Typography>
            <IconButton onClick={() => onDispatchAction()} sx={realEstateListIconStyles} aria-label="searcheable real estate" color="secondary">
                <SearchIcon fontSize={Size.Large} />
            </IconButton>
        </>
    );

    return (
        <Paper sx={realEstateListPaperStyles} elevation={3}>
            {hasItems ? (
                <Box>
                    {items.map((item: RealEstate) => (
                        <Box key={item.id} sx={realEstateListItemStyles}>
                            <RealEstateListItem item={item} />
                            <Delimiter sx={realEstateListDelimiterStyles} />
                        </Box>
                    ))}
                    {hasPagination && <Pagination page={page} count={numberOfPages} variant="outlined" shape="rounded" onChange={onChangePage} />}
                </Box>
            ) : (
                <Box sx={realEstateListLoaderStyles}>{getLoaderBlock()}</Box>
            )}
        </Paper>
    );
};
export default RealEstateList;
