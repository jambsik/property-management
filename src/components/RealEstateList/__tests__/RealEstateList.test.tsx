import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { basicRealEstateItemsDataMock } from '../../../__tests__/__fixtures__/realEstateDataMock';
import { Pagination } from '../../../constants/Pagination';
import RealEstateList from '../RealEstateList';
import { addTestProviders } from '../../../__tests__/__utils__/renderHelper';
import { useTranslation } from 'react-i18next';

describe('Real Estate List component test', () => {
    const onDispatchActionMock = jest.fn();
    const { t } = useTranslation();

    const pageObject = {
        pageOne: 1,
        pageTwo: 2,
        loaderBlock: t('realEstate.list.subTitle'),
    };

    const paginationTestOpts = {
        selector: 'button',
    };

    describe('Visualization Behaviour', () => {
        it('Should render component without errors', () => {
            const component = addTestProviders(
                <RealEstateList items={basicRealEstateItemsDataMock} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={2} />
            );

            const { container } = render(component);
            expect(container).toMatchSnapshot();
        });

        it('Should not render loader block if has items', () => {
            const component = addTestProviders(
                <RealEstateList items={basicRealEstateItemsDataMock} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={2} />
            );

            const { queryByText } = render(component);

            expect(queryByText(pageObject.loaderBlock)).not.toBeInTheDocument();
        });

        it("Should render loader block if hasn't items", () => {
            const component = addTestProviders(<RealEstateList items={[]} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={2} />);

            const { queryByText } = render(component);

            expect(queryByText(pageObject.loaderBlock)).toBeInTheDocument();
        });

        it('Should render items data if has items list', () => {
            const component = addTestProviders(
                <RealEstateList items={basicRealEstateItemsDataMock} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={2} />
            );
            const itemsTestOptions = {
                selector: 'p',
                exact: true,
            };

            const itemDataToFind = basicRealEstateItemsDataMock[0];

            const { queryAllByText, queryByText } = render(component);
            const streetElements = queryAllByText(itemDataToFind.street, itemsTestOptions);
            const yearBuiltElements = queryAllByText(itemDataToFind.yearbuilt, itemsTestOptions);
            const salePriceElement = queryByText(itemDataToFind.saleprice, itemsTestOptions);

            expect(streetElements.length).toBeGreaterThan(0);
            expect(yearBuiltElements.length).toBeGreaterThan(0);
            expect(salePriceElement).toBeInTheDocument();
            expect(salePriceElement).toHaveTextContent(`${itemDataToFind.saleprice}`);
        });
    });

    describe('Pagination Behaviour', () => {
        it('Should not display the pagination component if the number of pages is one', () => {
            const component = addTestProviders(
                <RealEstateList items={basicRealEstateItemsDataMock} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={1} />
            );

            const { queryByLabelText } = render(component);

            expect(queryByLabelText(pageObject.pageOne, paginationTestOpts)).not.toBeInTheDocument();
        });

        it('Should render pagination component if number of pages is greater than one', () => {
            const component = addTestProviders(
                <RealEstateList items={basicRealEstateItemsDataMock} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={24} />
            );

            const { queryByText } = render(component);

            expect(queryByText(pageObject.pageOne, paginationTestOpts)).toBeInTheDocument();
        });

        it('Should dispatch action with page after click in pagination component page', () => {
            const component = addTestProviders(
                <RealEstateList items={basicRealEstateItemsDataMock} page={Pagination.DefaultPage} onDispatchAction={onDispatchActionMock} numberOfPages={24} />
            );

            const { queryByText } = render(component);

            const pageElement = queryByText(pageObject.pageTwo, paginationTestOpts);

            expect(pageElement).toBeInTheDocument();

            fireEvent.click(pageElement!);

            expect(onDispatchActionMock).toHaveBeenCalledTimes(1);
            expect(onDispatchActionMock).toHaveBeenCalledWith(pageObject.pageTwo);
        });
    });
});
