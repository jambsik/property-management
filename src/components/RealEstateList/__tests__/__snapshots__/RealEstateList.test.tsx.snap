// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Real Estate List component test Visualization Behaviour Should render component without errors 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-amy87t-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-i3pbo"
      >
        <a
          href="/real-estate/1"
        >
          <div
            class="MuiBox-root css-759u60"
          >
            <img
              alt="home-image"
              class="css-1ninaiz"
              src="/images/home.svg"
            />
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-4gcdpv-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.street
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  Pave
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yrsold
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2008
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yearbuilt
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2003
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.saleprice
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  208500
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.garagecars
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.kitchenabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  1
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.bedroomabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  3
                </p>
              </div>
            </div>
          </div>
        </a>
        <div
          class="css-h0smfe"
        />
      </div>
      <div
        class="MuiBox-root css-i3pbo"
      >
        <a
          href="/real-estate/2"
        >
          <div
            class="MuiBox-root css-759u60"
          >
            <img
              alt="home-image"
              class="css-1ninaiz"
              src="/images/home.svg"
            />
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-4gcdpv-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.street
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  Pave
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yrsold
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2007
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yearbuilt
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  1976
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.saleprice
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  181500
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.garagecars
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.kitchenabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  1
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.bedroomabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  3
                </p>
              </div>
            </div>
          </div>
        </a>
        <div
          class="css-h0smfe"
        />
      </div>
      <div
        class="MuiBox-root css-i3pbo"
      >
        <a
          href="/real-estate/3"
        >
          <div
            class="MuiBox-root css-759u60"
          >
            <img
              alt="home-image"
              class="css-1ninaiz"
              src="/images/home.svg"
            />
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-4gcdpv-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.street
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  Pave
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yrsold
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2008
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yearbuilt
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2001
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.saleprice
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  223500
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.garagecars
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.kitchenabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  1
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.bedroomabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  3
                </p>
              </div>
            </div>
          </div>
        </a>
        <div
          class="css-h0smfe"
        />
      </div>
      <div
        class="MuiBox-root css-i3pbo"
      >
        <a
          href="/real-estate/4"
        >
          <div
            class="MuiBox-root css-759u60"
          >
            <img
              alt="home-image"
              class="css-1ninaiz"
              src="/images/home.svg"
            />
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-4gcdpv-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.street
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  Pave
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yrsold
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  2006
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.yearbuilt
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  1915
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.saleprice
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  140000
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.garagecars
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  3
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.kitchenabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  1
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
                >
                  realEstate.filters.bedroomabvgr
                  :
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
                >
                  3
                </p>
              </div>
            </div>
          </div>
        </a>
        <div
          class="css-h0smfe"
        />
      </div>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-selected MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              2
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-previousNext css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
